omod/src/main/java/org/openmrs/module/pihcore/page/controller/account/AccountPageController.java

```diff
@@ -77,43 +84,36 @@ public String post(@MethodParam("getAccount") @BindParams PihAccountDomainWrappe
                        @SpringBean("userService") UserService userService,
                        @SpringBean("adminService") AdministrationService administrationService,
                        @SpringBean("providerManagementService") ProviderManagementService providerManagementService,
-                       @SpringBean("accountValidator") AccountValidator accountValidator, PageModel model,
+                       @SpringBean("pihAccountValidator") PihAccountValidator pihAccountValidator,
+                       PageModel model,
                        HttpServletRequest request) {
 
         // manually bind userEnabled (since checkboxes don't submit anything if unchecked));
         account.setUserEnabled(userEnabled);
 
-        accountValidator.validate(account, errors);
-
-        String email = account.getEmail();
-        if (StringUtils.isNotBlank(email)) {
-            if (!EmailValidator.getInstance().isValid(email)) {
-                errors.rejectValue("email", "error.email.invalid");
-            }
-            else {
-                User existingUser = userService.getUserByUsernameOrEmail(email);
-                if (existingUser != null && !existingUser.equals(account.getUser())) {
-                    if (email.equalsIgnoreCase(existingUser.getEmail())) {
-                        errors.rejectValue("email", "emr.account.error.emailAlreadyInUse");
-                    }
-                }
-            }
-        }
+        pihAccountValidator.validate(account, errors);
 
         if (!errors.hasErrors()) {
             try {
+
+                if (!Context.hasPrivilege(PihEmrConfigConstants.PRIVILEGE_APP_EMR_SYSTEM_ADMINISTRATION)) {
+                    String msg = messageSourceService.getMessage("require.unauthorized");
+                    throw new APIException(msg);
+                }
+
                 accountService.saveAccount(account);
                 request.getSession().setAttribute(EmrConstants.SESSION_ATTRIBUTE_INFO_MESSAGE,
                         messageSourceService.getMessage("emr.account.saved"));
                 request.getSession().setAttribute(EmrConstants.SESSION_ATTRIBUTE_TOAST_MESSAGE, "true");
 
-                return "redirect:/pihcore/account/manageAccounts.page";
+                return "redirect:/pihcore/account/account.page?personId="+account.getPerson().getPersonId();
             } catch (Exception e) {
```

Comment:
Returns to view mode rather than back to the accounts list, when an account is saved.